onnxscript/function_libs/torch_aten/ops/linalg.py

```diff
@@ -0,0 +1,246 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+"""torch.ops.aten operators under the `linalg` module.
+
+- No inplace operators.
+- All functions should not have the script() decorator. This is because
+    we want to delay the compilation of the function.
+"""
+from __future__ import annotations
+
+from typing import Optional, Sequence
+
+from onnxscript import TensorType
+
+
+def aten_linalg_cholesky(self: TensorType, *, upper: bool = False) -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_cholesky_ex(
+    self: TensorType, *, upper: bool = False, check_errors: bool = False
+) -> tuple[TensorType, TensorType]:
+    raise NotImplementedError()
+
+
+def aten_linalg_cond(self: TensorType, p: Optional[float] = None) -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_cross(self: TensorType, other: TensorType, *, dim: int = -1) -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_det(A: TensorType) -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_diagonal(
+    A: TensorType, *, offset: int = 0, dim1: int = -2, dim2: int = -1
+) -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_eig(self: TensorType) -> tuple[TensorType, TensorType]:
+    raise NotImplementedError()
+
+
+def aten_linalg_eigh(self: TensorType, UPLO: str = "L") -> tuple[TensorType, TensorType]:
+    raise NotImplementedError()
+
+
+def aten_linalg_eigvals(self: TensorType) -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_eigvalsh(self: TensorType, UPLO: str = "L") -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_householder_product(input: TensorType, tau: TensorType) -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_inv(A: TensorType) -> TensorType:
+    raise NotImplementedError()
+
+
+def aten_linalg_inv_ex(
+    A: TensorType, *, check_errors: bool = False
```

Comment:
Currently, onnxscript doesn't handle things like keyword-only-arguments. Should probably decide what to support (specifically, in terms of what's required, what's not, distinguishing ONNX value-parameters vs. attributes, etc.)

Reply:
I think it’s fine to not support them, presuming the order of the attributes doesn’t matter and can be supplied by name?

Reply:
Yes, that's the convention being followed currently: specify actual inputs positionally and attribute-values as name=value pairs.